src/main/java/seedu/linkedout/logic/parser/SearchCommandParser.java

```diff
@@ -15,18 +20,57 @@
     /**
      * Parses the given {@code String} of arguments in the context of the SearchCommand
      * and returns a SearchCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public SearchCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_JOB);
+
+        if (!anyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_JOB)
```

Comment:
This expression can be a bit over-complicated, if possible can split the 3 conditions into 3 different booleans with corresponding names. 

Written in pseudocode:

```suggestion
        boolean nameForCond1 = !anyPrefixes ...
        boolean nameForCond2 = !argMultimap...
        boolean nameForCond3 = !args.isEmpty()...
        if (cond1 || cond2 || cond3)
```